src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -343,6 +348,7 @@ def run_query(
                 if not kwargs.get("_statement_params"):
                     kwargs["_statement_params"] = {}
                 kwargs["_statement_params"]["SNOWPARK_SKIP_TXN_COMMIT_IN_DDL"] = True
+            statement_params = kwargs.get("_statement_params")
```

Comment:
Should we inline this line to the `notify_debug_info_listeners`? Or move it right above the `notify` line if we want easier breakpoint experience. Want to avoid changes in between as much as possible. I could see that it is getting the same object, but just in case we decided to do something like recreating `kwargs`.